src/app/stores/reducer.ts

```diff
@@ -728,21 +729,32 @@ export function environmentReducer(
 
     case ActionTypes.ADD_ROUTE_RESPONSE: {
       const newRouteResponse: RouteResponse = action.routeReponse;
-
+      const newUUID = uuid();
+      const activeRouteResponseUUID = state.activeRouteResponseUUID === newRouteResponse.uuid ?
+        newUUID : newRouteResponse.uuid;
       newState = {
         ...state,
-        activeRouteResponseUUID: newRouteResponse.uuid,
+        activeRouteResponseUUID,
```

Comment:
This part should not be changed. As for other actions, the new object (here a route response) should come with a new UUID from the service (like for env duplication https://github.com/mockoon/mockoon/blob/master/src/app/services/environments.service.ts#L154). Please create a new `renewRouteResponseUUID` method in the data service for this or modify the UUID in the schema builder method you created.

Reply:
The reason for creating a new UUID here was to be able to insert the duplicated route response just beneath the active route response. I didn't want to create another reducer which mostly looks like the `ActionTypes.ADD_ROUTE_RESPONSE` reducer.

I can't find a way to keep the logic in `ActionTypes.ADD_ROUTE_RESPONSE` reducer and not modifying the `action.routeResponse`.

But it seems to me creating a new reducer is the best and cleaniest way to go for this change.

What is your opinion? Is creating a new reducer overkill? Or maybe is it too much look-alike code?

Reply:
I think that a new action is overkill when it's basically the same than ADD_ROUTE_RESPONSE. 
But you can modify this action, like it has been done for the ADD_ROUTE action. You will see that for this one the parameter afterUUID has been added in order to insert the duplicated route after its source. 
I think for duplicating route responses it should be done the same way. Not because it's the best one but so various implementations of similar features are the same.